pipes-and-filters/README.md

```diff
@@ -0,0 +1,145 @@
+﻿# Pipes and Filters pattern example
+
+This directory contains an example of the [Pipes and Filters cloud design pattern](https://learn.microsoft.com/azure/architecture/patterns/pipes-and-filters).
+
+This example contains three filters that perform processing on an image. The three filters are combined into a pipe; the output of one filter is passed as the input to the next. The filters are implemented as separate Function App functions and storage queue is the pipe. While this pattern shows the Function App functions being co-located under the same Function App for simplicity, they could very well be deployed individually to support more per-filter autonomy/scale.
+
+The sample takes a source image, resizes it (first filter), adds a watermark (second filter), then publishes it to a output destination (third filter). In this sample, the order of the first two filters could be reversed as they are not dependent on each other. The pipe is Azure Storage queues.
+
+## :rocket: Deployment guide
+
+Install the prerequisites and follow the steps to have deploy and run a example of the Pipes and Filters pattern.
+
+### Prerequisites
+
+- A resource group in [Azure subscription](https://azure.com/free) under which you can create resources.
+- [Git](https://git-scm.com/downloads)
+- [.NET 8 SDK](https://dotnet.microsoft.com/download/dotnet/8.0)
+- [Azure Functions Core Tools](https://learn.microsoft.com/azure/azure-functions/functions-run-local#install-the-azure-functions-core-tools)
+- [Azure CLI](https://learn.microsoft.com/cli/azure/install-azure-cli)
+
+### Steps
+
+1. Clone this repository to your workstation and navigate to the working directory.
+
+   ```shell
+   git clone https://github.com/mspnp/cloud-design-patterns
+   cd pipes-and-filters
+   ```
+
+1. Log into Azure and set subscription information.
+
+   ```azurecli
+   az login
+   az account set -s <Name or ID of subscription>
+
+   RESOURCE_GROUP_NAME=<Existing Azure resource group name>
```

Comment:
I started this way, and then switched because it then requires soemone to have permission to create a resource group (which -- shouldn't be that big of a deal).  But since that was your inclination as well, I went back to that.  Had to change the pre-reqs to match.  You'll find this change in [`ac2a0bf` (#118)](https://github.com/mspnp/cloud-design-patterns/pull/118/commits/ac2a0bf1d4a2dfadfcd14f629d3a3f26b97a6f50)